### ISSUE TITLE
Cosine Decay Not Properly Enforced

### ISSUE THREAD

**Issue by ahatamiz** [NVIDIA-NeMo/RL#1532]
When passing the following hyper-parameters , one expects to see a cosine decay profile:

```
scheduler:
      start_weight_decay: ${policy.megatron_cfg.optimizer.weight_decay}
      end_weight_decay: ${policy.megatron_cfg.optimizer.weight_decay}
      weight_decay_incr_style: "cosine"
      lr_decay_style: "cosine"
      lr_warmup_iters: 25
      lr_warmup_init: 1e-7
```

However, although warmup is properly enforced, learning rate seems to remain constant. Please see the figure below. 

<img width="1209" height="904" alt="Image" src="https://github.com/user-attachments/assets/76d51cc3-2a75-4a1b-9d9a-8a50797a0295" />

We appreciate if there is a solution for actual enforcement of a cosine decay profile.

**Comment by snowmanwwg** [comment]
@ahatamiz - does this solve your problem?
